src/main/java/org/wso2/carbon/module/swiftiso20022/validation/rules/custom/MT103ArrayFieldValidationRule.java

```diff
@@ -0,0 +1,114 @@
+
+import java.util.List;
+
+/**
+ * Validate fields with arrays of strings.
+ * Default case is a repetitive field.
+ * If the field is a multiline field use the constructor with the boolean
+ *
+ */
+public class MT103ArrayFieldValidationRule implements ValidationRule {
+
+    private final List<ValidatorContext> contexts;
+
+    private final boolean isMultilineField;
+
+    // for repetitive fields
+    public MT103ArrayFieldValidationRule(List<ValidatorContext> contexts) {
+        this.contexts = contexts;
+        this.isMultilineField = false;
+    }
+
+    // For multiline fields
+    public MT103ArrayFieldValidationRule(List<ValidatorContext> contexts, boolean isMultilineField) {
+        this.contexts = contexts;
+        this.isMultilineField = isMultilineField;
+    }
+
+    @Override
+    public ValidationResult validate(JSONObject payload) {
+        for (ValidatorContext context : this.contexts) {
+
+            // validation happens only if the key is present
+            if (payload.has(context.getFieldName())) {
+                JSONArray values = payload.getJSONArray(context.getFieldName());
+
+                // if the key is present there should be at least one value
+                if (values.length() == 0) {
+                    return new ValidationResult(ConnectorConstants.ERROR_CODE_INVALID_PARAM,
+                            String.format(ConnectorConstants.ERROR_PARAMETER_EMPTY, context.getFieldDisplayName()));
+                }
+
+                if (isMultilineField) {
+                    int linesAllowed = (int) context.getProperty(MT103Constants.LINES_ALLOWED_KEY);
+
+                    if (values.length() > linesAllowed) {
+                        return new ValidationResult(ConnectorConstants.ERROR_CODE_INVALID_PARAM,
+                                String.format(ConnectorConstants.ERROR_LINE_COUNT,
+                                        context.getFieldDisplayName(), linesAllowed));
+                    }
+                }
+
+                // each value is validated
+                for (int i = 0; i < values.length(); i++) {
+                    String stringValue = values.getString(i);
+
+                    // value cannot be blank
+                    if (stringValue.isBlank()) {
+                        return new ValidationResult(ConnectorConstants.ERROR_CODE_INVALID_PARAM,
+                                String.format(isMultilineField ? ConnectorConstants.ERROR_LINE_EMPTY
+                                                : ConnectorConstants.ERROR_REPETITION_EMPTY,
+                                        ++i, context.getFieldDisplayName()));
+                    }
+
+                    // value length cannot be longer than defined length
+                    if (isMultilineField) {
+
+                        // multiline field line is compared against maximum line length
+                        if (stringValue.length() > MT103Constants.MT103_TEXT_LINE_LENGTH) {
```

Comment:
Shall we pass the length in the context FieldLength and validate ?

Reply:
Changed and refactored in commit 4e687a247135e6bc9d43274b4f9f97756cf64480